src/storage/btree.c

```diff
@@ -24877,7 +24878,16 @@ btree_range_scan_descending_fix_prev_leaf (THREAD_ENTRY * thread_p, BTREE_SCAN *
   /* Pages are still linked. */
 
   /* Fix current page too. */
-  bts->C_page = pgbuf_fix (thread_p, &bts->C_vpid, OLD_PAGE, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
+  if (bts->force_restart_from_root)
+    {
+      bts->C_page = pgbuf_fix (thread_p, &bts->C_vpid, OLD_PAGE, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
```

Comment:
Why we don't always use pgbuf_fix_old_and_check_repl_desync ?
The flag force_restart_from_root can be set for various reasons (btree changed by other transaction, page ahead  due to replication).
By always using pgbuf_fix_old_and_check_repl_desync, we can catch the "page ahead error", clear it  and make sure to retry, otherwise the scan is stopped with error.

Reply:
My understanding in this situation is that the page is "most probably" still in the page buffer. Thus, it will be in-sync with replication when re-fixed again after having been unfixed at the beginning of the function.
To be perfectly safe, I would still `pgbuf_fix_old_and_check_repl_desync` and, if `C_page` remains null, assert `assert (er_errid () != ER_PAGE_AHEAD_OF_REPLICATION);`